universal-application-tool-0.0.1/test/controllers/admin/AdminProgramControllerTest.java

```diff
@@ -0,0 +1,104 @@
+import models.Program;
+import org.junit.Before;
+import org.junit.Test;
+import play.mvc.Http.Request;
+import play.mvc.Http.RequestBuilder;
+import play.mvc.Result;
+import play.test.Helpers;
+import repository.WithResettingPostgresContainer;
+import views.html.helper.CSRF;
+
+public class AdminProgramControllerTest extends WithResettingPostgresContainer {
+
+  private AdminProgramController controller;
+
+  @Before
+  public void setup() {
+    controller = app.injector().instanceOf(AdminProgramController.class);
+  }
+
+  @Test
+  public void listWithNoPrograms() {
+    Result result = controller.list();
+    assertThat(result.status()).isEqualTo(OK);
+    assertThat(result.contentType()).hasValue("text/html");
+    assertThat(result.charset()).hasValue("UTF-8; charset=utf-8");
+    assertThat(contentAsString(result)).contains("Programs");
+  }
+
+  @Test
+  public void list_returnsPrograms() {
+    insertProgram("one");
+    insertProgram("two");
+
+    Result result = controller.list();
+
+    assertThat(result.status()).isEqualTo(OK);
+    assertThat(contentAsString(result)).contains("one");
+    assertThat(contentAsString(result)).contains("two");
+  }
+
+  @Test
+  public void newOne_returnsExpectedForm() {
+    Request request = addCSRFToken(Helpers.fakeRequest()).build();
+
+    Result result = controller.newOne(request);
+
+    assertThat(result.status()).isEqualTo(OK);
+    assertThat(contentAsString(result)).contains("Create a new Program");
+    assertThat(contentAsString(result)).contains(CSRF.getToken(request.asScala()).value());
+  }
+
+  @Test
+  public void create_returnsNewProgramInList() {
+    RequestBuilder requestBuilder =
+        Helpers.fakeRequest()
+            .bodyForm(
+                ImmutableMap.of("name", "New Program", "description", "This is a new program"));
+
+    Result result = controller.create(requestBuilder.build());
+
+    assertThat(result.status()).isEqualTo(OK);
+    assertThat(contentAsString(result)).contains("New Program");
+    assertThat(contentAsString(result)).contains("This is a new program");
+  }
+
+  @Test
+  public void create_includesNewAndExistingProgramsInList() {
+    insertProgram("Existing One");
+    RequestBuilder requestBuilder =
+        Helpers.fakeRequest()
+            .bodyForm(
+                ImmutableMap.of("name", "New Program", "description", "This is a new program"));
+
+    Result result = controller.create(requestBuilder.build());
+
+    assertThat(result.status()).isEqualTo(OK);
+    assertThat(contentAsString(result)).contains("Existing One");
+    assertThat(contentAsString(result)).contains("New Program");
+    assertThat(contentAsString(result)).contains("This is a new program");
+  }
+
+  private static void insertProgram(String name) {
+    Program program = new Program(name, "description");
+    program.save();
+  }
+
+  /**
+   * For our HTTP responses, we set the charset to "UTF-8; charset=utf-8", which is not
```

Comment:
question: Is this something we can / need to fix in the [`contentType` method](https://github.com/seattle-uat/universal-application-tool/blob/main/universal-application-tool-0.0.1/app/views/BaseHtmlView.java#L62)? I know @kendrickt mentioned this at standup, but I didn't fully follow the issue and what we might do to resolve it.

Reply:
Yes, thank you! That was the issue - the `contentType` method is returning both content type and charset, which means the charset is in there twice. Removing it allows us to use Play's `Helper#contentAsString` method successfully

Reply:
Glad to hear that resolves it!! 🎉